miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppHttpWebView.kt

```diff
@@ -0,0 +1,45 @@
+package com.rakuten.tech.mobile.miniapp.display
+
+import android.content.Context
+import com.rakuten.tech.mobile.miniapp.MiniAppInfo
+import com.rakuten.tech.mobile.miniapp.MiniAppScheme
+import com.rakuten.tech.mobile.miniapp.js.MiniAppMessageBridge
+import com.rakuten.tech.mobile.miniapp.navigator.MiniAppNavigator
+import com.rakuten.tech.mobile.miniapp.permission.MiniAppCustomPermissionCache
+import kotlin.random.Random
+
+internal class MiniAppHttpWebView(
+    context: Context,
+    miniAppTitle: String,
+    val appUrl: String,
+    miniAppMessageBridge: MiniAppMessageBridge,
+    miniAppNavigator: MiniAppNavigator?,
+    hostAppUserAgentInfo: String,
+    miniAppWebChromeClient: MiniAppWebChromeClient = MiniAppWebChromeClient(context, miniAppTitle),
+    miniAppCustomPermissionCache: MiniAppCustomPermissionCache
+) : MiniAppWebView(
+    context,
+    "",
+    MiniAppInfo.empty(),
+    miniAppMessageBridge,
+    miniAppNavigator,
+    hostAppUserAgentInfo,
+    miniAppWebChromeClient,
+    miniAppCustomPermissionCache
+) {
+    init {
+        miniAppScheme = MiniAppScheme.schemeWithCustomUrl(appUrl)
+        miniAppId = "custom${Random.nextInt(0, Int.MAX_VALUE)}" // some id is needed to handle permissions
```

Comment:
Could you help us to understand what kind of permissions need to handle?

Reply:
@khairul-alam-licon I think this is so the custom permissions will be cached to a random ID. Is that right @imaginaris ?

We don't need the permissions to be cached for URLs. But I think this solution is okay because each time a URL is launched it will be assigned a different ID to use for caching. Trying to disable caching would probably need some more major refactoring. 

But a disadvantage is that we end up with a lot of permissions cached for random IDs which we don't need. Is it possible to delete these from the cache when the mini app is closed? For example something like this:

```
    override fun onDetachedFromWindow() {
        super.onDetachedFromWindow()
        
        miniAppCustomPermissionCache.removeId(miniAppId) // you would need to add a method like this
    }
```

Reply:
@corycaywood Loading by url does not proceed with `MiniAppDownloader` and `MiniAppStatus` does not cache anything.
We need a dummy miniapp id to ensure other parts are working properly (such as retrieve access token).
-> @imaginaris  solution is okay.

Reply:
I have another concern when users launch a same url twice e.g. `https://www.google.com` it will produce different ids for the same url. Should we care about it? 

Also, if we need to generate this random id, so we could probably match with the format what backend provides.

e.g. `miniAppId = UUID.randomUUID().toString()`

Reply:
```suggestion
        miniAppId = UUID.fromString(appUrl).toString()
```

Reply:
Ah, I just realized when I try a normal url e.g. `https://www.google.com/` using `UUID.fromString(appUrl).toString()`, it returns the following error:

```
java.lang.IllegalArgumentException: Invalid UUID string: https://www.google.com/
```
So, maybe we can attempt with `UUID.randomUUID().toString()` instead.